### DISCUSSION TITLE
How to use make lead function in sqlalchemy?

### DISCUSSION THREAD

**Discussion by thinksource** [sqlalchemy/sqlalchemy#6508] [Usage Questions]
Hi I just want to lead function like

```
select  a.*,lead(a.localnumber,{}) OVER (ORDER BY a.objectid) next_number from xbtpitn a where a.subtype in ('LAT', 'TEL')
```

Does sqlalchmey can do similar query? like have lead or lag function?

**Comment by CaselIT** [comment]
Hi,

You should be able to use the func attribute to create that function. See the docs https://docs.sqlalchemy.org/en/14/core/functions.html?highlight=func#module-sqlalchemy.sql.functions
